source/elements/oneMKL/source/domains/spblas/data_types/set_coo_matrix_data.rst

```diff
@@ -0,0 +1,209 @@
+                                 sycl::buffer<indexType, 1>           &col_ind,
+                                 sycl::buffer<dataType, 1>            &val);
+
+   }
+
+.. container:: section
+
+   .. rubric:: Template parameters
+
+   dataType
+      See :ref:`supported template types<onemkl_sparse_supported_types>`. Can be
+      a different type than what was used when creating the ``matrix_handle_t``.
+
+   indexType
+      See :ref:`supported template types<onemkl_sparse_supported_types>`. Can be
+      a different type than what was used when creating the ``matrix_handle_t``.
+
+.. container:: section
+
+   .. rubric:: Input parameters
+
+   queue
+      The SYCL command queue which will be used for SYCL kernels execution.
+
+   smhandle
+      Handle already initialized with :ref:`onemkl_sparse_create_coo_matrix`.
+
+   num_rows
+      Number of rows of the provided data ``val``. Must be at least 0.
+
+   num_cols
+      Number of columns of the provided data ``val``. Must be at least 0.
+
+   nnz
+      Number of non-zero entries in the matrix (which may include explicit
+      zeros). Must be at least 0.
+
+   index
+      Indicates how input arrays are indexed. The possible options are described
+      in :ref:`onemkl_enum_index_base` enum class.
+
+   row_ind
+      Buffer of length ``nnz`` containing the row indices in ``index``-based
+      numbering. Refer to :ref:`onemkl_sparse_coo` format for detailed
+      description of ``row_ind``.
+
+   col_ind
+      Buffer of length ``nnz`` containing the column indices in ``index``-based
+      numbering. Refer to :ref:`onemkl_sparse_coo` format for detailed
+      description of ``col_ind``.
+
+   val
+      Buffer of length ``nnz`` containing non-zero elements (and possibly
+      explicit zeros) of the input matrix. Refer to :ref:`onemkl_sparse_coo`
+      format for detailed description of ``val``.
+
+.. container:: section
+
+   .. rubric:: Throws
+
+   This function shall throw the following exceptions if the associated
+   condition is detected. An implementation may throw additional
+   implementation-specific exception(s) in case of error conditions not covered
+   here.
+
+   | :ref:`oneapi::mkl::device_bad_alloc<onemkl_exception_device_bad_alloc>`
+   | :ref:`oneapi::mkl::invalid_argument<onemkl_exception_invalid_argument>`
+   | :ref:`oneapi::mkl::unimplemented<onemkl_exception_unimplemented>`
+   | :ref:`oneapi::mkl::uninitialized<onemkl_exception_uninitialized>`
+   | :ref:`oneapi::mkl::unsupported_device<onemkl_exception_unsupported_device>`
+
+.. _onemkl_sparse_set_coo_matrix_data_usm:
+
+set_coo_matrix_data (USM version)
+---------------------------------
+
+.. rubric:: Syntax
+
+.. code-block:: cpp
+
+   namespace oneapi::mkl::sparse {
+
+       template <typename dataType, typename indexType>
+       void set_coo_matrix_data (sycl::queue                          &queue,
+                                 oneapi::mkl::sparse::matrix_handle_t smhandle,
```

Comment:
In my initial comment I should have said `sparse_matrix_handle_t` instead of `sparse_matrix_t`. 

I can see that `matrix_handle_t` might reasonably be assumed to be sparse in the sparse API, as you say. But then you also have `sparse_matrix_handle_t` used in some parts of the API (eg. [here](https://github.com/Rbiessy/oneAPI-spec/blob/romain/sparse_api/source/elements/oneMKL/source/domains/spblas/data_types/set_matrix_property.rst#set_matrix_property-1)). I'm finding the distinction confusing.

Reply:
Sorry, the occurrences of `sparse_matrix_handle_t` were a mistake and were renamed to `matrix_handle_t` in https://github.com/oneapi-src/oneAPI-spec/pull/522/commits/8a8f6759c7e3677668b1554f299d4f59b7328d9e